### PR TITLE
[PBA-2767] Implement continuos playback on discovery end screen

### PR DESCRIPTION
**Pull request by brol1dev**
First code version for the continuos playback on the discovery end screen. **Don't merge it yet**, I will add some more code to make it more robust, but on this version you can see it in action.

### CODE REVIEW

**Comment by brol1dev** [comment]
You can merge now if you want :)

File: sdk/OoyalaSkinSDK/discoveryPanel.js [hunk 1]
```diff
@@ -69,6 +81,30 @@ var DiscoveryPanel = React.createClass({
...
+      showCircularStatus: true,
+    });
+  },
+
+  updateTimer: function(row) {
+    if (this.state.currentCounterVal == 0) {
+      this.onRowSelected(row);
+    } else {
+      var self = this;
+      setTimeout(function() {
```
Comment (lental) [inline sdk/OoyalaSkinSDK/discoveryPanel.js:101]: This is only slightly concerning to me - It is the UI causing things to happen.  I don't have the best idea of if there is a better way to do this, but I'll voice my concern at least, and keep reading

File: sdk/OoyalaSkinSDK/discoveryPanel.js [hunk 2]
```diff
@@ -99,6 +135,16 @@ var DiscoveryPanel = React.createClass({
...
   },
 
+  renderCircularStatus: function() {
+    return (
+      <CircularStatus
+        onPress={() => this.onStatusPressed()}
+        total={this.state.counterLimit}
+        current={this.state.currentCounterVal}
+        thickness={2}
+        diameter={44} />);
```
Comment (lental) [inline sdk/OoyalaSkinSDK/discoveryPanel.js:145]: we have to worry about hard coded values at some point, maybe not at this point

File: sdk/OoyalaSkinSDK/widgets/CircularStatus.js [hunk 3]
```diff
@@ -17,10 +17,11 @@ var CircularStatus = React.createClass({
     current: React.PropTypes.number,
     total: React.PropTypes.number,
     diameter: React.PropTypes.number,
-    thickness: React.PropTypes.number
+    thickness: React.PropTypes.number,
```
Comment (lental) [inline sdk/OoyalaSkinSDK/widgets/CircularStatus.js:20]: why did you add thickness?

File: sdk/OoyalaSkinSDK/widgets/ResponsiveList.js [hunk 4]
```diff
@@ -67,20 +67,25 @@ var ResponsiveList = React.createClass({
 
   renderSlice: function(slice: object, i: number) {
     var sliceStyle = this.props.horizontal ? styles.column : styles.row;
+
+    renderItem = this.renderItem;
```
Comment (lental) [inline sdk/OoyalaSkinSDK/widgets/ResponsiveList.js]: is there any reason you need this as a variable?

File: sdk/OoyalaSkinSDK/discoveryPanel.js (hunk 1 above)
Comment (brol1dev) [inline sdk/OoyalaSkinSDK/discoveryPanel.js:101]: Ok, let me think of another way of implementing this.

File: sdk/OoyalaSkinSDK/widgets/CircularStatus.js (hunk 3 above)
Comment (brol1dev) [inline sdk/OoyalaSkinSDK/widgets/CircularStatus.js:20]: I just added the comma at the end.

Comment (lental) [inline sdk/OoyalaSkinSDK/widgets/CircularStatus.js:20]: whoops, true

File: sdk/OoyalaSkinSDK/widgets/ResponsiveList.js (hunk 4 above)
Comment (brol1dev) [inline sdk/OoyalaSkinSDK/widgets/ResponsiveList.js]: I did that because if we call 'this' inside the slice.map context, we can't call the defined functions from the ResponsiveList. So I make the renderItem a variable so I can access it within slice.map.

File: sdk/OoyalaSkinSDK/discoveryPanel.js (hunk 1 above)
Comment (lental) [inline sdk/OoyalaSkinSDK/discoveryPanel.js:101]: i wouldn't say dump it yet.  I think it's okay, and if you have any new interesting ideas, let me know before coding it up.  it's more of a curious case to think about

Comment (brol1dev) [inline sdk/OoyalaSkinSDK/discoveryPanel.js:101]: I'm thinking of having a module (maybe residing within the CircularStatus.js) that would send an event to the UI (discoveryPanel.js) that a new second thick in the clock happened. So the UI is only responsible of updating the UI by receiving this event. But I really don't know how hard it would be to implement it.

File: sdk/OoyalaSkinSDK/widgets/ResponsiveList.js (hunk 4 above)
Comment (lental) [inline sdk/OoyalaSkinSDK/widgets/ResponsiveList.js]: is it possible to add the 'var' to the front of it?  it feels like this is leaking a little out of scope

Comment (brol1dev) [inline sdk/OoyalaSkinSDK/widgets/ResponsiveList.js]: Oh yea, totally. I missed that one.

File: sdk/OoyalaSkinSDK/discoveryPanel.js (hunk 2 above)
Comment (brol1dev) [inline sdk/OoyalaSkinSDK/discoveryPanel.js:145]: I agree, but where could be a good place for they to live? Within constants? In the skin-config?

File: sdk/OoyalaSkinSDK/discoveryPanel.js (hunk 1 above)
Comment (lental) [inline sdk/OoyalaSkinSDK/discoveryPanel.js:101]: First, do you need to stop the timeout at any point, or does this go on forever?

I think its fine how it is currently.  I think i may be overcomplicating UI. @Matthew-C , do you have any opinions on this? using setTimeout in javascript for timers, and then using it to autoplay the next video?

**Comment by lental** [comment]
Gonna let it in for now. We can discuss changes going forward. thanks
